Automattic/jetpack
# Social: Add the reshare endpoint to the Publicize package.

#### Changes proposed in this Pull Request:

Like the other endpoints in the package that are used in the editor,
this proxies the request to WPCOM.

Which endpoint to use is different depending on whether it's being used from WPCOM and Jetpack or Jetpack Social. This change also adds the ability to configure the endpoint URL path via the editor's initial state.

With this in place and #25993 merged, then it should be possible to reshare a post via the block editor in Jetpack Social.

#### Other information:


#### Jetpack product discussion
Making resharing available to all is discussed here p1HpG7-gqr-p2

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Install Jetpack Social on a test site and check out this PR
* Add the `publicize-share-limit` feature flag
* Check that a published post can be reshared from the block editor
* Similarly check that resharing is still possible with the Jetpack plugin. Both with a paid plan and with the feature flag.
